data/types/blood/glucose/selfmonitored/selfmonitored.go

```diff
@@ -66,18 +67,26 @@ func (s *SelfMonitored) Normalize(normalizer data.Normalizer) {
 		normalizer = normalizer.WithMeta(s.Meta())
 	}
 
-	s.Glucose.Normalize(normalizer)
+	if normalizer.Origin() == structure.OriginExternal {
+
+		rawUnits := pointer.CloneString(s.Units)
+		rawValue := pointer.CloneFloat64(s.Value)
+		s.SetRawUnitsAndValue(rawUnits, rawValue)
+
+		s.Units = dataBloodGlucose.NormalizeUnits(rawUnits)
+		s.Value = dataBloodGlucose.NormalizeValueForUnits(rawValue, rawUnits)
+	}
```

Comment:
I *think* you can get rid of the Clone here if you just reverse the normalization of units and value, so this can become:
```suggestion
	if normalizer.Origin() == structure.OriginExternal {
	    s.SetRawUnitsAndValue(s.Units, s.Value)
		s.Value = dataBloodGlucose.NormalizeValueForUnits(s.Value, s.Units)
		s.Units = dataBloodGlucose.NormalizeUnits(s.Units)
	}
```